src/App.tsx

```diff
@@ -1,24 +1,168 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, {
+  useContext,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import {
+  StateContext,
+} from './components/TodoContext';
+import { TodoList } from './components/TodoList';
+import { FILTER, ACTIONS } from './utils/enums';
+import { addTodo, deleteTodo, getTodos } from './api/todos';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  const [query, setQuery] = useState('');
+  const { state, dispatch } = useContext(StateContext);
+  const [newTodo, setNewTodo] = useState<null | Todo>(null);
+
+  function deleteAll() {
+    state.list.forEach(todo => {
+      if (todo.completed) {
+        deleteTodo(todo.id)
+          .then(() => getTodos(11384)
+            .then(res => {
+              dispatch({ type: ACTIONS.SET_LIST, payload: res });
+            }));
+      }
+    });
+  }
+
+  function handleFilter(trigger: string) {
+    dispatch({ type: ACTIONS.SORT, payload: trigger });
+  }
+
+  const initialTodo = {
+    id: 0,
+    title: query,
+    userId: 11384,
+    completed: false,
+  };
+
+  function handleEnter(e: React.KeyboardEvent<HTMLInputElement>) {
+    if (e.nativeEvent.code === 'Enter') {
+      e.preventDefault();
+      if (!e.target.value.split('').every(element => element === ' ')) {
+        setNewTodo(initialTodo);
+        addTodo(initialTodo)
+          .then(() => getTodos(11384)
+            .then(res => {
+              dispatch({ type: ACTIONS.SET_LIST, payload: res });
+              setNewTodo(null);
+            }));
```

Comment:
```suggestion
          .then(() => getTodos(11384))
          .then(res => {
            dispatch({ type: ACTIONS.SET_LIST, payload: res });
            setNewTodo(null);
          });
```
we can simplify it by chaining

Reply:
I agree with your suggestion but eslint not )
<img width="1280" alt="Screenshot 2023-09-13 at 13 43 30" src="https://github.com/mate-academy/react_todo-app-add-and-delete/assets/20277989/46e787e3-e608-4805-aa39-a6727c1af1ab">